src/core/calculate_pf.jl

```diff
@@ -0,0 +1,601 @@
+    node_to_idx = Dict(bt=>i for (i, bt) in enumerate(indexed_nodes))
+
+    N = length(indexed_nodes)
+    Y = spzeros(Complex{Float64}, N, N)
+    for (ns, y_prim) in ns_yprim
+        inds = [node_to_idx[n] for n in ns]
+        Y[inds,inds] .+= y_prim
+    end
+
+    node_fixed_idx = [i for i in 1:N if ntype[indexed_nodes[i]]==FIXED]
+    node_other_idx = setdiff(1:N, node_fixed_idx)
+
+    fnode_to_idx = Dict(node=>idx for (idx,node) in enumerate(indexed_nodes[node_fixed_idx]))
+    vnode_to_idx = Dict(node=>idx for (idx,node) in enumerate(indexed_nodes[node_other_idx]))
+
+    Yf = Y[node_other_idx, node_fixed_idx]
+    Yv = Y[node_other_idx, node_other_idx]
+
+    fixed_nodes = [indexed_nodes[i] for i in 1:N if ntype[indexed_nodes[i]]==FIXED]
+    Uf = fill(NaN+im*NaN, length(fixed_nodes))
+
+    for (i,(b,t)) in enumerate(fixed_nodes)
+        vm_t = Dict(data_math["bus"]["$b"]["terminals"].=>data_math["bus"]["$b"]["vm"])
+        va_t = Dict(data_math["bus"]["$b"]["terminals"].=>data_math["bus"]["$b"]["va"])
+        Uf[i] = vm_t[t]*exp(im*va_t[t])
+    end
+
+    Yv_LU = factorize(Yv)
+
+    return PowerFlowData(data_math, ntype, cc_ns_func_pairs, indexed_nodes, node_to_idx, fnode_to_idx, vnode_to_idx, fixed_nodes, Uf, Yf, Yv_LU)
+end
+
+
+"""
+    _get_v(
+      pfd::struct,
+      Vp::Vector,
+      n::Vector
+    )
+
+Calculates the voltage from PowerFlowData struct.
+"""
+function _get_v(pfd::PowerFlowData, Vp::Vector{Complex{Float64}}, n::Tuple{Int64, Int64})
+    if pfd.ntype[n] == GROUNDED
+        return 0.0+im*0.0
+    elseif pfd.ntype[n] == FIXED
+        return pfd.Uf[pfd.fnode_to_idx[n]]
+    else
+        return Vp[pfd.vnode_to_idx[n]]
+    end
+end
+
+"""
+    compute_pf(
+      data_math::Dict,
+      v_start::Dict,
+      max_iter::Int,
+      stat_tol::Float,
+      verbose::Bool
+    )
+
+Computes native power flow and outputs the result dict.
+"""
+function compute_pf(data_math::Dict{String, Any}; v_start::Union{Dict{<:Any,<:Any},Missing}=missing, max_iter::Int=100, stat_tol::Float64=1E-8, verbose::Bool=true)
+    if !ismultinetwork(data_math)
+        nw_dm = Dict("0"=>data_math)
+    else
+        nw_dm = data_math["nw"]
+    end
+
+    sol = Dict{String, Any}()
+    time_build = Dict{String, Any}()
+    time_solve = Dict{String, Any}()
+    time_post = Dict{String, Any}()
+    Yf_matrix = Dict{String, Any}()
+    Yv_matrix = Dict{String, Any}()
+    status = Dict{String, Any}()
+    its = Dict{String, Any}()
+    stat = Dict{String, Any}()
+
+    for (nw, dm) in nw_dm
+        if ismissing(v_start)
+            add_start_vrvi!(dm)
+            v_start = _bts_to_start_voltage(dm)
+        end
+
+
+        time_build[nw] = @elapsed pfd = PowerFlowData(dm, v_start)
+
+        time_solve[nw] = @elapsed (Uv, status[nw], its[nw], stat[nw]) = _compute_Uv(pfd, max_iter=max_iter, stat_tol=stat_tol)
+
+        time_post[nw] = @elapsed sol[nw] = build_solution(pfd, Uv)
+
+        Yf_matrix[nw] = pfd.Yf
```

Comment:
let's discuss if we really return everything listed here (think: powermodels doesn't return the JuMP model either)